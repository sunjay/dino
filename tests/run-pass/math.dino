fn main() {
    // Need to specify the type in the literal so the
    // method call isn't ambiguous
    let value = -34int + 43;
    print_int(value);
    value = value + -2;
    print_int(value);
    value = -1int + value + (value - 94);
    print_int(value + 1);
}

fn foo(x: int) -> int {
    if x > 2 {
        return x * 2 + 5 / x;
    }

    -x - 2
}

fn precedence(x: int, y: int, z: int, w: int) -> int {
    x * (2int + y) / z - (x - y) % w + x
}
