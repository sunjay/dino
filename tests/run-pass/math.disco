fn main() {
    let value: int = add_int(-34, 43);
    print_int(value);
}
