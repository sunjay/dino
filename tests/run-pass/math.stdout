9
7
-80
20
